### ISSUE TITLE
Amethyst suddenly stops working

### ISSUE THREAD

**Issue by chhetrisushil** [ianyh/Amethyst#392]
There is no specific scenario but amethyst sudden stops working. I can see the process running and using the shortcuts shows the information of which layout is selected but to no effect e.g. I'm trying to switch to fullscreen mode but nothing changes (I can see the text "fullscreen" on the display).

Not sure if anyone else is facing the same issue.

**Comment by ianyh** [comment]
That doesn't sound good. What version of Amethyst?

**Comment by m4rw3r** [comment]
Could it be related to the Accessibility setting (ie. System Preferences -> Security & Privacy -> Privacy -> Accessibility) for the stops working part? I have found that I have had to manually check Amethyst there a few times to allow it to move and resize windows, though that could have been related to me building my own binary.

As for amethyst not recognizing windows, I usually switch spaces back and forth for that. But yes, I have seen similar issues with firefox (ie. if Firefox is launched before Amethyst, Amethyst will not detect it unless it is relaunched or spaces are switched).

**Comment by ianyh** [comment]
The toggling of accessibility permissions should only be necessary if code signature changes. So building your own binary goes from code signature to no code signature. Subsequently going to a production version goes from no code signature to code signature. There was a problem going from 0.8 to 0.9 where the signatures somehow changed, so some people had to toggle. That was painful. If you've only been updating within the app or downloading signed binaries from here or on the website that shouldn't be the issue.

**Comment by weeezes** [comment]
Changing the accessibility permissions like @m4rw3r suggested fixed this for me. Problem started after upgrading from 0.8 to 0.9 as @ianyh mentioned.

**Comment by chhetrisushil** [comment]
same for me the problem started after upgrading to 0.9.10 version (which I'm currently using). And as far as the accessibility permission goes I've already given Amethyst permissions but if you mean I need to keep verifying it I believe that still qualifies as a bug (correct me if I'm wrong). Still I gave the accessibility thing a try but nope it did not really work for me. And also once I relaunched Amethyst it worked fine for sometime but the problem appeared again, had to restart the system, after which I haven't faced the issue yet (laptop has been running for almost 24 hours after that last restart).

**Comment by ianyh** [comment]
@chhetrisushil it is totally a bug. I want to, at some point, deal with the accessibility permissions thing in-app. (You can do it with sudo permission.) Next time Amethyst stops working can you check the running applications to see if one of them is hanging? Like bringing up the menu with `cmd+alt+esc` and looking for anything in red with (Not responding) after it.

**Comment by chhetrisushil** [comment]
@ianyh sure will do

**Comment by chhetrisushil** [comment]
it happened again (just now to be exact) and I did as u asked (forgot to take the snapshot though, sorry about that). In the menu none of the programmes were hanging i.e. nothing was in red every application was working fine.

**Comment by ianyh** [comment]
Are you running any Java-based applications? Like Eclipse or IntelliJ?

**Comment by chhetrisushil** [comment]
No. I do have 3 windows of MacVim running at a time and 9 windows of iterm, 2 window of chrome (with around 4-5 tabs each), Skype, Slack, HipChat, Preview, Atom, Outlook, Lync that's about it.

**Comment by phlod** [comment]
Next time this happens, try disconnecting your USB devices.
This happens to me periodically, and by chance I noticed when I unplugged my MS Natural Keyboard, that Amethyst immediately started working again and rearranged my windows.

**Comment by nakaumin** [comment]
I have same issue on both MacbookAir 2015 (High Sierra) and Mac mini 2012 (Sierra).

Amethyst suddenly stops working almost every 15-30 minutes (very cumbersome!).
My certain recovery workaround is to kill Amethyst process and restart Amethyst.
(I assigned keyboard shortcut using BTT to run the shellscript for the workaround)

On my environments, apps below are running:
Alfred, BTT, Evernote, Karabiner-elements, Docker, Dropbox and HazeOver.

**Comment by vlaadbrain** [comment]
@ianyh Hi, I'm Experiencing this issue (using Amethyst version 0.12.0 57) i just installed Eclipse and it becomes unmanaged :( i have to restart amethyst for it to continue being managed.

**Comment by astrostl** [comment]
Love the tool when it works, while having the same experience. I am relaunching Amethyst dozens of times per day because it randomly and regularly fails to pick up changes.

**Comment by mibzman** [comment]
Yeah I'm running into this also.

**Comment by florianbellmann** [comment]
How is this issue open 5 years? I just recently discovered amethyst and love it. But the regular necessary restarts really make me think to switch away from it again. 
From a normal user perspective this should be the most important issue to fix, right? :)

**Comment by astrostl** [comment]
FWIW I backed down to using Magnet. I'd love to revisit Amethyst if/when this is resolved!

**Comment by florianbellmann** [comment]
Magnet doesn't automatically adjust your panes if a window is closed/opened, right? That's really something I need. For simple windowmanagement I used https://github.com/rxhanson/Rectangle so far.

**Comment by florianbellmann** [comment]
Does anyone have at least a hint to where the issue might be? Somewhere to start looking for a fix? :)

**Comment by phlod** [comment]
Just what I posted in 2016.  When this happened to me, unplugging my USB keyboard, or occasionally unplugging all my USB devices would make it suddenly start working again.

**Comment by quaelin** [comment]
I've been living with this issue for a while and would love to see it fixed... but a very simple workaround works for me: _swipe to a different desktop, wait 1 second, then swipe back._

**Comment by florianbellmann** [comment]
I figured out that amethyst works much better if I only minimize windows (cmd+m) than hide then (cmd+h). Now it's only rarely that I have to restart Amethyst. 

I also found this https://github.com/ianyh/Amethyst/issues/741
and it should be fixed here https://github.com/ianyh/Amethyst/issues/662

Unfortunately it's still happening.
